test/server/live-test.js

```diff
@@ -0,0 +1,329 @@
+const expect               = require('chai').expect;
+const Promise              = require('pinkie');
+const path                 = require('path');
+const createTestCafe       = require('../../lib/index');
+const Controller           = require('../../lib/live/controller');
+const FileWatcher          = require('../../lib/live/file-watcher');
+const LiveModeRunner       = require('../../lib/live/test-runner');
+const LiveModeBootstrapper = require('../../lib/live/bootstrapper');
+
+const fileName1 = path.resolve('test/server/data/test-suites/live/testfile1.js');
+const fileName2 = path.resolve('test/server/data/test-suites/live/testfile2.js');
+const fileName3 = path.resolve('test/server/data/test-suites/live/testfile3.js');
+const fileName4 = path.resolve('test/server/data/test-suites/live/testfile4.js');
+
+const externalModulePath = path.resolve('test/server/data/test-suites/live/module.js');
+
+class FileWatcherMock extends FileWatcher {
+    constructor (files) {
+        super(files);
+    }
+
+    addFile (file) {
+        if (file.indexOf('node_modules') > -1)
+            return;
+
+        this.files = this.files || [];
+
+        this.files.push(file);
+    }
+}
+
+let errors = [];
+
+class ControllerMock extends Controller {
+    constructor (runner) {
+        super(runner);
+
+        this.logger = {
+            intro:          () => {},
```

Comment:
You can use [lodash.noop](https://lodash.com/docs/4.17.11#noop) instead of `() => {}`